test/src/main/java/org/axonframework/test/utils/RecordingMessageMonitor.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.test.utils;
+
+import org.axonframework.messaging.Message;
+import org.axonframework.monitoring.MessageMonitor;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A message monitor implementation that records the processing results of messages in the provided
+ * {@link MessageMonitorReport} instances. It supports logging the processing outcomes if logging is enabled. This
+ * monitor tracks whether messages were successfully processed, failed, or ignored and stores this information in the
+ * provided reports.
+ * <p>
+ * The class provides two constructors: one allows configuring a single report instance while enabling logging, and the
+ * other accepts a list of reports, defaulting logging to disabled.
+ * <p>
+ * The monitor provides functionality to handle messages as they are ingested and uses a callback to record the
+ * processing result for each message.
+ */
+public class RecordingMessageMonitor implements MessageMonitor<Message> {
+
+    final MessageMonitorReport report;
+    private final Logger logger;
```

Comment:
nit:

Logger can be a static field - like in other places.

Reply:
Well, here I set it dynamically to the given context ... so static wont work. But as you said: logging was merely a way of debugging while I build this, so I think I will drop logger, context and isLoggingEnabled completely.